core/src/docs/upgrade.md

```diff
@@ -1,3 +1,7 @@
+# Upgrade to v0.41
+
+There is no public API and raw API changes.
```

Comment:
Java binding's API changed.

Reply:
This is our rust core's upgrade guide. We do not need to add stuff about Java binding here.
Java binding has its own upgrade guide: https://github.com/apache/incubator-opendal/blob/main/bindings/java/upgrade.md

Reply:
Cool, please add in java's upgrade notes.

Reply:
It's already done.